### PR TITLE
Add the studio field

### PR DESCRIPTION
**Pull request by OmarIthawi**
Inspired by the [edX Zoom XBlock](https://github.com/edx/edx-zoom/blob/37c323ae93265937bf60abb92657318efeec96c5/edx_zoom/utils.py#L6-L11).

<kbd>image</kbd>

### CODE REVIEW

File: course_access_groups/utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from django.utils.translation import ugettext as _
+
+log = logging.getLogger(__name__)
+
+
+def add_to_course_fields():
+    """
+    Add the `course_access_groups` field to the course.
+
+    This is monkey patching hence the appropriate warning message.
```
Comment (johnbaldwin) [inline course_access_groups/utils.py]: Can you describe a bit more on what this does exactly? from a quick scan of the docstring, all I understand is that this is a monkepatch hack for course access groups, but have no idea what it is supposed to do

File: course_access_groups/utils.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+from __future__ import absolute_import, unicode_literals
```
Comment (johnbaldwin) [inline course_access_groups/utils.py]: Can we use a different module name for this? One of my colleagues and I had discussions over time with using files called "util". Then encourage dumping functions that do not quite match other places. He convinced me that avoiding "utils" is a good practice for maintainability. 

And the function `add_course_fields` is not a utility method so should not be in a 'utils' module anyway, if we had an already existing `utils` module.

Instead, would you please call this module  `course_access_groups.monkeypatch` so readers are very clear this is workaround hacking?

**Comment by johnbaldwin** [comment]
@OmarIthawi cool effort on this. Have comments though and the build is failing

File: course_access_groups/utils.py (hunk 2 above)
Comment (OmarIthawi) [reply course_access_groups/utils.py]: Great idea! Will do so. Thanks John!

File: course_access_groups/utils.py (hunk 1 above)
Comment (OmarIthawi) [reply course_access_groups/utils.py]: Will reword to make it clearer.

**Comment by OmarIthawi** [comment]
Thanks @johnbaldwin, I've addressed your notes. I think now tests will pass. Please let me know if you have additional notes.

File: course_access_groups/utils.py (hunk 1 above)
Comment (johnbaldwin) [reply course_access_groups/utils.py]: Thank you!

File: course_access_groups/utils.py (hunk 2 above)
Comment (johnbaldwin) [reply course_access_groups/utils.py]: Thank you!

File: course_access_groups/monkeypatch.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+from django.utils.translation import ugettext as _
+
+log = logging.getLogger(__name__)
+
+
+def add_to_course_fields():
+    """
+    Add the `course_access_groups` field to the CourseFields class.
+
+    This makes every course has the new `course_access_groups` field without the need to hard code such logic for
```
Comment (johnbaldwin) [inline course_access_groups/monkeypatch.py]: What is in the `course_access_groups` field? Looks like a list. What is in the list?

Comment (OmarIthawi) [reply course_access_groups/monkeypatch.py]: @johnbaldwin it's the field below, please see lines 29-37 it has documentation embedded:

```python
CourseFields.course_access_groups = List(
    display_name=_('Course Access Groups'),
    help=_(
        'The IDs for the Course Access Groups which this course is available to. Enter the IDs in the '
        'following JSON array format: ["employees", "customers", "university"]'
    ),
    default=[],
    scope=Scope.settings
)
```

**Comment by OmarIthawi** [comment]
> Looks like Travis is still failing. I tried to run tox locally but am getting the error

Please worry about tests failing as long as Travis catches them. I think it's more valuable to focus on the other part of the code reviews.

> ```
> py36-django111 create: /Users/jbaldwin/work/appsembler/repos/course-access-groups/.tox/py36-django111
> ERROR: InterpreterNotFound: python3.6
> py36-django20 create: /Users/jbaldwin/work/appsembler/repos/course-access-groups/.tox/py36-django20
> ERROR: InterpreterNotFound: python3.6
> ```
> 
> I have Python 3.6.10 installed with Pyenv. Do I need a different version of 3.6?

Yes. I think `tox` doesn't know about the `pyenv` pythons and it looks only for system-wide installations such as:

```
$ whereis python
python: /usr/bin/python2.7-dbg-config /usr/bin/python /usr/bin/python3.6 /usr/bin/python3.6m-config /usr/bin/python3.6m /usr/bin/python2.7-dbg /usr/bin/python3.6-config /usr/bin/python2.7-config /usr/bin/python2.7 /usr/lib/python3.6 /usr/lib/python3.8 /usr/lib/python3.7 /usr/lib/python2.7 /etc/python /etc/python3.6 /etc/python2.7 /usr/local/lib/python3.6 /usr/local/lib/python2.7 /usr/include/python3.6 /usr/include/python2.7_d /usr/include/python3.6m /usr/include/python2.7 /usr/share/python /opt/az/bin/python3.6 /opt/az/bin/python3.6m-config /opt/az/bin/python3.6m /opt/az/bin/python3.6-config /usr/share/man/man1/python.1.gz
```

**Comment by johnbaldwin** [comment]
Thanks Omar. I'll have to work on my local python environment to get tox to pass with Python 3.6. Meanwhile, LGTM, Approved!

**Comment by OmarIthawi** [comment]
Thanks John and Maxi!

**Pull request merged**